docs/core/event_handler/api_gateway.md

```diff
@@ -859,6 +859,33 @@ your development, building, deployment tooling need to accommodate the distinct
 * **Slower safe deployments**. Safely deploying multiple functions require coordination — AWS CodeDeploy deploys and verifies each function sequentially. This increases lead time substantially (minutes to hours) depending on the deployment strategy you choose. You can mitigate it by selectively enabling it in prod-like environments only, and where the risk profile is applicable.
     * Automated testing, operational and security reviews are essential to stability in either approaches.
 
+**Example**
+
+Consider a simplified micro function structured REST API that has two routes:
+
+* `/users` - an endpoint that will return all users of the application on `GET` requests
+* `/users/<id>` - an endpoint that looks up a single users details by ID on `GET` requests
+
+Each endpoint will be it's own Lambda function that is configured as a [Lambda integration](https://docs.aws.amazon.com/apigateway/latest/developerguide/getting-started-with-lambda-integration.html){target="_blank"}. This allows you to set different configurations for each lambda (memory size, layers, etc.)
+
+=== "`/users` Endpoint"
+    ```python
+    --8<-- "examples/event_handler_rest/src/micro_function_all_users_route.py"
+    ```
+
+=== "`/users/<id>` Endpoint"
+    ```python
+    --8<-- "examples/event_handler_rest/src/micro_function_user_by_id_route.py"
+    ```
+
+=== "Micro Function Example SAM Template"
+    ```yaml
+    --8<-- "examples/event_handler_rest/sam/micro_function_template.yaml"
+    ```
+
+???+ note
+    You can see some of the downsides in this example such as some code reuse. If set up with proper build tooling, the `User` class could be shared across functions.
```

Comment:
I really like this explanation, but do you think we can add something here about using the Lambda Layer to avoid code reuse in common/shared functions?